### ISSUE TITLE
Getting error in vue-ts-mode--range-settings

### ISSUE THREAD

**Issue by wenzlawski** [8uff3r/vue-ts-mode#8]
When running I get the error

```
defvar: Wrong type argument: treesit-query-p, :local
```

Looking at the documentation, it does not seem like `treesit-range-rules` includes a keyword for local. Am I missing something here?

**Comment by wenzlawski** [comment]
Seems like this was introduced in Emacs 30

**Comment by oLd-Y** [comment]
Same error. Is there any solutions?

**Comment by wenzlawski** [comment]
Besides updating to Emacs 30, no. The functionality this is relying on is introduced in v30, and as treesitter package is integrated into Emacs you can't update it independently.

Alternatively you can disable local treesitter fragments by commenting out the offending code.

**Comment by 8uff3r** [comment]
Hi @wenzlawski 
Are you still experiencing the issue after updating to Emacs 30?

**Comment by wenzlawski** [comment]
No. The thing is the code seems to be incompatible with Emacs <30, since it relies on functionality that does not exist in lower versions. Specifically that is the ability to render local tree-sitter fragments. Here I am referring to the `:local t` statements in line 172 and 177. 

See this thread in the mailiing list: https://mail.gnu.org/archive/html/emacs-devel/2023-09/msg00607.html

**Comment by aspiers** [comment]
In that case I'm confused why the package states a requirement on emacs 29.1:

https://github.com/8uff3r/vue-ts-mode/blob/5ec5bb317b80ce394e156c61b7b9c63996382a68/README.org?plain=1#L2

https://github.com/8uff3r/vue-ts-mode/blob/5ec5bb317b80ce394e156c61b7b9c63996382a68/vue-ts-mode.el#L8

**Comment by 8uff3r** [comment]
The minimum required version for Emacs has been set to 30, unfortunately version 29 has many problems with treesitter.
